ui/appui-react/src/appui-react/hooks/useElementSize.ts

```diff
@@ -0,0 +1,111 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import React, { useEffect, useRef, useState } from "react";
+
+import debounce from "lodash/debounce.js";
+import { useDeepCompareMemo } from "./useDeepCompareMemo.js";
+
+/**
+ * Breakpoint sizes.
+ */
+export type BreakpointSize = "xs" | "sm" | "md" | "lg" | "xl" | "2xl";
+
+/**
+ * Breakpoints interface.
+ */
+export interface Breakpoints extends Record<BreakpointSize, number> {
+  xs: number;
+  sm: number;
+  md: number;
+  lg: number;
+  xl: number;
+  "2xl": number;
+}
+
+/**
+ * Return type of the useElementSize hook.
+ */
+export interface UseElementSizeReturn {
+  ref: React.RefObject<HTMLDivElement | null>;
+  size: BreakpointSize;
+  dimension: { width: number; height: number };
+}
+
+export const defaultBreakpoints: Breakpoints = {
+  xs: 0,
+  sm: 300,
+  md: 600,
+  lg: 1024,
+  xl: 1280,
+  "2xl": 1536,
+};
+
+/**
+ * Custom hook to get the size and dimensions of an HTML element.
+ *
+ * @param {Partial<Breakpoints>} [breakpoints] - Optional breakpoints to override the default breakpoints.
+ * @param {object} [options] - Optional options object.
+ * @param {number} [options.debounceMs] - Debounce time in milliseconds for the resize event.
+ * @returns {UseElementSizeReturn} The ref to attach to the element, the current breakpoint size, and the dimensions of the element.
+ */
+export function useElementSize(
```

Comment:
Let's avoid adding this sort of API, as it's not related to layouts and is a separate feature. Have you considered something like https://developer.mozilla.org/en-US/docs/Web/CSS/Guides/Containment/Container_queries